x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_logs_step/fix_logs_step.tsx

```diff
@@ -32,6 +34,25 @@ const i18nTexts = {
       defaultMessage: 'Resolve deprecation issues and verify your changes',
     }
   ),
+  apiCompatibilityNoteTitle: i18n.translate(
+    'xpack.upgradeAssistant.overview.apiCompatibilityNoteTitle',
+    {
+      defaultMessage: 'Apply API compatibility headers (optional)',
+    }
+  ),
+  apiCompatibilityNoteBody: (docLink: string) => (
+    <FormattedMessage
+      id="xpack.upgradeAssistant.overview.apiCompatibilityNoteBody"
+      defaultMessage="We advise bringing your log count down to zero to guarantee your upgrade will be free of issues. However, it can be challenging to be certain that all requests have been fixed, and by applying the compatibility header to all requests provides an additional safety net. {learnMoreLink}."
```

Comment:
I don't think it's a safety net, I think it's an either-or thing: you address the deprecation and test, or you include a compatibility header. If you've made modifications to be compatible with 8.0 and include headers that request 7.x compatibility, I think things will break. AFAIK, you theoretically could choose to fix some things, but not others--it's set on a per-request basis. 

There isn't currently a good place to link to--I'll fix that & provide the URL.

```suggestion
      defaultMessage="We recommend resolving deprecation issues before upgrading.  However, if you need to enable an application to continue to function without making changes, you can include API compatibility headers in your requests. Learn more{learnMoreLink}."
```

Reply:
Thanks for pointing that out! I've fixed the copy with your suggestion and waiting for your ticket to be merged so I can add the link here.

Reply:
@debadair The guidance I received from @jakelandis was that we advise users to apply the REST API compatibility header to all API requests, but it’s not guaranteed to address any problems -- it might help, it might not. We advise users to bring their log count down to zero, because that’s the only way to guarantee that their upgrade will be free of issues. However, it can be challenging to be certain that all requests have been fixed, and by applying the compatibility header to all requests users give themselves an additional safety net.

Jake, could you please confirm or clarify?

Reply:
@cjcenizal - confirm. Most importantly "We advise users to bring their log count down to zero, because that’s the only way to guarantee that their upgrade will be free of issues. "

Since compatiblity is only API compatibility and not a pure versioned API there can times when compatiblity if applied may not work. For example, if your request includes a type (as in mapping types) without compatiblity you will get an error, likey a parsing or missing error, but if you use REST API compatiblity it will still allow that request to be processed, but will ignore the requested type since in 8.0 there is no such thing as mapping types.  In some other cases compatiblity simply can't be applied and will result in a more friendly error message. REST API compatiblity sweet spot is when the REST API changed in a way where there is a 1:1 replacement of something old to something new... in those cases it map the old -> new pretty easily. 

@debadair - I have a todo to kickstart the conversation around the public documentation for this feature. We need to be careful to identify the right balance of documenting the value and caveats of using this.